From d3f2763c2a16b09d347b12c36ddc31185f0fe4d3 Mon Sep 17 00:00:00 2001
From: Don Zickus <dzickus@redhat.com>
Date: Tue, 16 Sep 2014 17:05:44 -0400
Subject: [usb] hub: debug message for failing to enable device

Message-id: <1410887463-20674-273-git-send-email-dzickus@redhat.com>
Patchwork-id: 95193
O-Subject: [RHEL7 PATCH 272/591] hub: debug message for failing to enable device
Bugzilla: 1110939
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>

https://bugzilla.redhat.com/show_bug.cgi?id=1110939

commit 938569eb75b329fee9f2634900ad7e12caf13fd2
Author: Oliver Neukum <oneukum@suse.de>
Date:   Thu Feb 27 10:57:10 2014 +0100

    hub: debug message for failing to enable device

    This error case isn't reported during enumeration.

    Signed-off-by: Oliver Neukum <oliver@neukum.org>
    Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/usb/core/hub.c b/drivers/usb/core/hub.c
index 60e7c98..9cf1114 100644
--- a/drivers/usb/core/hub.c
+++ b/drivers/usb/core/hub.c
@@ -4112,8 +4112,12 @@ hub_port_init (struct usb_hub *hub, struct usb_device *udev, int port1,
 
    did_new_scheme = true;
    retval = hub_enable_device(udev);
-   if (retval < 0)
+   if (retval < 0) {
+    dev_err(&udev->dev,
+     "hub failed to enable device, error %d\n",
+     retval);
     goto fail;
+   }
 
 #define GET_DESCRIPTOR_BUFSIZE 64
    buf = kmalloc(GET_DESCRIPTOR_BUFSIZE, GFP_NOIO);
-- 
1.7.1